Dynatrace/dynatrace-operator
# In `csidriver` pod, split `driver` container to `server` and `provisioner`

# Description

Currently, in `csidriver` pod, `driver` container contains logic of serving csi endpoint, as well as provisioning agent images.

The latter has spiky characteristics of resource consuming (cpu, mem) because it downloads agents from DT tenant, unpacks it and cleans it up if needed.

If `driver` container hits resource limits, it is killed and csi driver is broken for pods where it's mounted.

This PR splits `driver` container to `server` and `provisioner`.
`server` does lightweight work like answering grpc requests.
`provisioner` downloads images, unpacks them and executes `CsiGarbageCollection`.

Resources limits and requests for `server` set to: `100m`/`50Mi`.

## How can this be tested?
Set up dynakube with csidriver enabled.
Run custom pod to be injected.
Check if server is available if provisioner exceeded resources limits and is restarted.

## Checklist
